okhttp-clients/src/main/java/com/palantir/conjure/java/okhttp/UrlSelectorImpl.java

```diff
@@ -113,84 +115,102 @@ private static String switchWsToHttp(String url) {
         }
     }
 
+    /**
+     * Attempt to redirect to the given redirectUrl, which could be a longer URL than just a base path, in which
+     * case it will first be matched to a baseUrl from {@link #baseUrls}.
+     */
     @Override
-    public Optional<HttpUrl> redirectTo(HttpUrl current, String redirectBaseUrl) {
-        return redirectTo(current, HttpUrl.parse(redirectBaseUrl));
+    public Optional<HttpUrl> redirectTo(HttpUrl current, String redirectUrl) {
+        return baseUrlFor(HttpUrl.parse(redirectUrl), baseUrls.get()).flatMap(baseUrl -> redirectTo(current, baseUrl));
```

Comment:
Basically pulled up this line from `redirectTo(HttpUrl, HttpUrl)`:
https://github.com/palantir/conjure-java-runtime/pull/1142/files#diff-97df878facbe38492456bc963582d9e8L122
because in all other invocations, the thing being passed in is guaranteed to be a base URL contained in `baseUrls`